src/js/utils/FormUtil.js

```diff
@@ -211,15 +216,47 @@ const FormUtil = {
   removePropID: function (definition, prop, id) {
     let fieldsToRemove = [];
     definition.forEach((field) => {
-      if (this.isFieldInstanceOfProp(prop, id, field)) {
+      if (this.isFieldInstanceOfProp(prop, field, id)) {
         fieldsToRemove.push(field);
```

Comment:
Do we want to remove this field even if a child (from the recursive check) is matched?